README.md

```diff
@@ -1,4 +1,174 @@
+    - `name`: name of the dimension,
+    - `size`: size (length) of the dimension,
+    - `original_temporal`, optional: boolean, should be `True` if this is the temporal dimension generated in the initial `load_collection` node.
+
+###### Methods
+
+* **`write`** ():
+
+    Returns the evalscript as a string.
+
+* **`determine_output_dimensions`** ()
+
+    Calculates the greatest possible dimensions of the output datacube, returning a list of dicts. Each element contains:
+    - `name`: name of the dimension,
+    - `size`: size (length) of the dimension,
+    - `original_temporal`, optional: boolean, `True` if this is the temporal dimension generated in the initial `load_collection` node.
+
+* **`set_output_dimensions`** (output_dimensions):
+
+    Setter for output dimensions. `output_dimensions` is a list of dicts. Each element contains:
+    - `name`: name of the dimension,
+    - `size`: size (length) of the dimension,
+    - `original_temporal`, optional: boolean, should be `True` if this is the temporal dimension generated in the initial `load_collection` node.
+
+* **`get_decoding_function`** ():
+
+    Returns a `decode_data` function. The data returned by the evalscript is encoded to contain the information about the datacube dimensions and has to be decoded to obtain the actual data in a ndarray format.
+    `decode_data` has the following parameters:
+    - `data`: the result of processing of the associated evalscript, it should be a three-dimensional array.
+    `decode_data` returns a multidimensional Python list.
+
+
+## Workflow
+
+1. Construct the process graph
+
+Load a file with `json.load` or generate a process graph using [openEO Python client](https://github.com/Open-EO/openeo-python-client).
+
+2. Run the conversion
+
+```python
+subgraph_evalscripts = convert_from_process_graph(process_graph)
+
+print(subgraph_evalscripts)
+>>> [{'evalscript': <pg_to_evalscript.evalscript.Evalscript object at 0x000001ABA779CA00>, 'invalid_node_id': None}]
+```
+
+In this example, the entire process graph could be converted to an evalscript, so we only have one entry.
+
+3. Fetch the data
+
+```python
+evalscript = subgraph_evalscripts[0]['evalscript'].write()
+print(evalscript)
+>>> "//VERSION=3 function setup(){ ..."
+```
+The evalscript string can now be used to process data on Sentinel Hub. [Sentinel Hub Python client](https://sentinelhub-py.readthedocs.io/en/latest/) makes it easy to do so. 
+
+4. Decode the fetched data
+
+```python
+# Get the decoding function for this evalsscript
+decoding_function = evalscript.get_decoding_function()
+# Pass the fetched data through the decoding function. 
+# The function expects a python list. If you're using Sentinel Hub Python client, the result might be a numpy array, so it has to be converted.
+decoded_data = decoding_function(fetched_data.tolist())
+print(decoded_data)
+>>> [[[1, 2, 3], [4, 5, 6], ... ]]
+```
+
+
+## Running notebooks
 
 ```
 pipenv install --dev
```

Comment:
Would add ```cd notebooks``` to the beginning of the "Start the notebooks" block (further down but it doesn't let me add the comment there)

Reply:
Was added, resolving.